app/controllers/games_controller.rb

```diff
@@ -18,9 +20,9 @@ def show
   end
 
   def join
-    @game = Game.find(params[:id]).available
-    @game.update_attributes(@game.black_player_id = current_user.id)
-    redirect_to join_games_path
+    @game = Game.find(params[:game_id])
+    @game.update_attributes(black_player_id: current_user.id)
+    redirect_to game_join_path
```

Comment:
The **redirect_to game_join_path** causes Rails to error out trying to GET /games/1/join.  What we want to redirect to instead, is the games show action using this route:

`game GET    /games/:id(.:format)           games#show`

Let's change this redirect method to:
`redirect_to game_path(@game)`

This way the redirect will go to /games/:id which is our desired result

Reply:
Updated - thanks!